sunpy/net/dataretriever/tests/test_goes_ud.py

```diff
@@ -9,15 +13,81 @@
 from sunpy.net.dataretriever.client import QueryResponse
 import sunpy.net.dataretriever.sources.goes as goes
 from sunpy.net.fido_factory import UnifiedResponse
+from sunpy.time.time import is_time_equal
 from sunpy.net import Fido
 from sunpy.net import attrs as a
 from sunpy.net.tests.strategies import goes_time
 from sunpy.time import parse_time, is_time_equal
 
 
-@pytest.fixture
-def LCClient():
-    return goes.XRSClient()
+LCClient = goes.XRSClient()
+
+
+def mock_query_object(start_date, end_date):
+    """
+    Creation of a QueryResponse object, and prefill some
+    downloaded data from goes.XRSClient().fetch(Time('20 ..)
+    """
+    # Create a mock QueryResponse object
+    map_ = {
+        'TimeRange' : TimeRange(parse_time(start_date), parse_time(end_date)),
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'nasa',
+        'instrument': 'goes',
+        'physobs': 'irradiance',
+        'provider': 'sdac'
+    }
+
+    st_datetime = datetime.strptime(start_date, '%Y/%m/%d')
+    ed_datetime = datetime.strptime(end_date, '%Y/%m/%d') + timedelta(days=1)
+    tr = map_['TimeRange']
+    time_range = tr.split(int((ed_datetime-st_datetime).days))
+
+    resp = QueryResponse.create(map_,
+                                LCClient._get_url_for_timerange(
+                                    map_['TimeRange']),
+                                time=time_range)
+    # Attach the client with the QueryResponse
+    resp.client = LCClient
+    return resp
+
+
+@pytest.mark.remote_data
+def test_fetch_working():
+    """
+    Tests if the online server for noaa is working.
+    Uses the url : ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt
+    """
+    qr1 = LCClient.search(Time('1995/06/03', '1995/06/05'),
+                          Instrument('XRS'))
+
+    # Mock QueryResponse object
+    mock_qr = mock_query_object('1995/06/03', '1995/06/05')
+
+    # Compare if two objects have the same attribute
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+
+    assert mock_qr.source == qr.source
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.url == qr.url
+
+    # Here the mock object contains `datetime.datetime` object by default,
+    # so no parsing required
+
+    sdate1 = mock_qr.time.start.value
+    sdate2 = datetime.strptime(qr.time.start.value, "%Y-%m-%dT%H:%M:%S.%f")
+
+    edate1 = mock_qr.time.end.value
+    edate2 = datetime.strptime(qr.time.end.value, "%Y-%m-%dT%H:%M:%S.%f")
+
+    # Use _is_time_equal to check if the dates are almost equal
+    assert is_time_equal(sdate2,sdate1)
```

Comment:
Is `mock_qr.time` and `qr.time`, `TimeRange` objects? If yes, you can do equality check directly on them.

Else you can check start of both with `is_time_equal`. Don't need conversion to datetime here.

Reply:
`is_time_equal` does not work for `TimeRange` objects.